tasties_app/models.py

```diff
@@ -1,3 +1,90 @@
-# from django.db import models
+from django.db import models
+from django.utils import timezone
+from django.contrib.auth.models import User
+from django.core.validators import MinValueValidator, MinLengthValidator, MaxValueValidator
 
-# Create your models here.
+
+class Category(models.Model):
+    category_name = models.CharField(max_length=16, unique=True)
+
+
+class Recipe(models.Model):
+    title = models.CharField(max_length=64, validators=[MinLengthValidator(1)], unique=True)
+    author_id = models.ForeignKey(User, on_delete=models.CASCADE)
+    categories = models.ManyToManyField(Category)
+    description = models.CharField(max_length=512, validators=[MinLengthValidator(1)])
+    directions = models.CharField(max_length=65536, validators=[MinLengthValidator(1)])
+    publication_date = models.DateField(default=timezone.now)
+    minutes_to_make = models.IntegerField(validators=[MinValueValidator(1)])
+    recipe_picture = models.ImageField(upload_to='recipe_pictures')
+
+
+class Ingredient(models.Model):
+    """
+    Ingredient Model - represents an ingredient in a recipe.
+    Fields:
+        UNIT_CHOICES - a list of valid measurement units
+        recipe_id - the recipe that the ingredient belongs to
+        amount - the amount of the ingredient
+        measurement_unit - the unit of measurement for the ingredient
+        description - a description of the ingredient
+    """
+
+    WHOLE = "WHOLE"
+    FLOZ = "FLOZ"
+    TSP = "TSP"
+    OZ = "OZ"
+    CUP = "CUP"
+    GRAM = "GRAM"
+    ML = "ML"
+    KG = "KG"
+    L = "L"
+    U = "U"
+    UNIT_CHOICES = ((WHOLE, 'Whole'),
+                    (FLOZ, 'Fluid Ounce'),
+                    (TSP, 'Teaspoon'),
+                    (OZ, 'Ounce'),
+                    (CUP, 'Cup'),
+                    (GRAM, 'Gram'),
+                    (ML, 'Milliliter'),
+                    (KG, 'Kilogram'),
+                    (L, 'Liter'),
+                    (U, 'Units'),
+                    )
+
+    recipe_id = models.ForeignKey(Recipe, on_delete=models.CASCADE)
+    amount = models.DecimalField(max_digits=10, decimal_places=2, validators=[MinValueValidator(0.0001)])
+    measurement_unit = models.CharField(max_length=8, choices=UNIT_CHOICES, default=WHOLE)
+    description = models.CharField(max_length=64, validators=[MinLengthValidator(1)])
+
+    def __str__(self):
+        return self.description + " " + str(self.amount) + " " + self.measurement_unit
+
+    def edit_info(self, amount, measurement_unit, description):
+        """
+        Edits the ingredient's information
+            amount = new amount value
+            measurement_unit = new measurement value
+            description = new description value
+        """
+        try:
+            self.amount = amount
```

Comment:
Do you think there might be a reason to put the contents of this try block under a transactions.atomic() context manager (with transactions.atomic():)?